prov/sm2/src/sm2.h

```diff
@@ -129,6 +136,28 @@ struct sm2_ep_name {
 	struct dlist_entry entry;
 };
 
+struct sm2_cmd_sar_hdr {
+	union {
+		uint32_t proto_flags;
+		size_t reserved;
+	};
+	void *context;
```

Comment:
The generic header has the context that the user wants to see in completion.  This context is the state that the SAR protocol needs.  perhaps it could be called "sar_context" or "protocol_context"

Reply:
done